source/common/http/utility.cc

```diff
@@ -296,6 +296,39 @@ std::string parseCookie(const HeaderMap& headers, const std::string& key,
   return EMPTY_STRING;
 }
 
+std::map<std::string, std::string> Utility::parseCookies(const RequestHeaderMap& headers) {
```

Comment:
This is very similar to the existing code in [source/common/http/utility.cc #L257-L282](https://github.com/envoyproxy/envoy/blob/main/source/common/http/utility.cc#L257-L282)

Could not use it as-is because that'd scan over cookies multiple times. Any suggestions on merging them together ?

Reply:
Thanks for the suggestion! Was able to get this working locally 👍 

I would prefer to do this change in a follow-up PR. I hope that's okay.

Reply:
Yes, a follow-up would be Ok. Just put a TODO note to the function.

Reply:
The other places using `parseCookieValue` would be benefit from this logic if you see how oauth2 filter use it.

https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/http/oauth2/filter.cc#L141-L143

I would suggest you to land the refactoring before this PR.

Reply:
Cool, I'll do that first